api/app_analytics/analytics_db_service.py

```diff
@@ -14,34 +14,109 @@
     FeatureEvaluationBucket,
     Resource,
 )
+from dateutil.relativedelta import relativedelta
 from django.conf import settings
 from django.db.models import Sum
 from django.utils import timezone
 
 from environments.models import Environment
 from features.models import Feature
+from organisations.models import Organisation
 
-ANALYTICS_READ_BUCKET_SIZE = 15
+from . import constants
 
 
 def get_usage_data(
-    organisation, environment_id=None, project_id=None
-) -> List[UsageData]:
+    organisation: Organisation,
+    environment_id: int | None = None,
+    project_id: int | None = None,
+    period: (
+        Literal[
```

Comment:
It's nicer to avoid defining the type inline here, i.e.

```python
UsageDataPeriod = Literal[
    constants.CURRENT_BILLING_PERIOD,
    constants.PREVIOUS_BILLING_PERIOD,
    constants._90_DAY_PERIOD,
]

...

def get_usage_data(
    ...
    period: UsageDataPeriod | None = None,
```

Reply:
Yeah I agree I'll do that when I update this PR.

Reply:
Moved this into its own module called `typing` with the other changes as noted above.

Reply:
Sorry if I'm annoying, but we have modules like this in the codebase and they're called `types`. Let's rename yours for consistency.

Reply:
Ah yes of course. Will update 👍🏼